release/RELEASE_PROCESS.md

```diff
@@ -2,48 +2,162 @@
-   - Edit the `docs/quickstart.md` file, updating the `RELEASE` variable in the `Get the latest release` section to the new release branch name.
-   - Edit the `scripts/setup_user_namespaces.sh` file, changing the `modelmesh_release` version.
-1. Submit your PR to the release branch and wait for it to merge.
-1. Update `docs/component-versions.md`, `docs/quickstart.md`, `docs/install/install-script.md`, and `scripts/setup_user_namespaces.sh` files in the main branch with the same versions as above, then submit this as a PR to `main`. Wait for this to merge.
-1. Generate release manifests:
-   - `kustomize build config/default > modelmesh.yaml`
-   - `kustomize build config/runtimes --load-restrictor LoadRestrictionsNone > modelmesh-runtimes.yaml`
-   - `cp config/dependencies/quickstart.yaml modelmesh-quickstart-dependencies.yaml`
-1. Generate config archive:
-   - Run one of the following depending on what version of `tar` you have. Be sure to replace the `update-me` below with the correct version:
-     - GNU tar: `RELEASE=update-me;tar -zcvf config-${RELEASE}.tar.gz config/ --transform s/config/config-${RELEASE}/`
-     - BSD tar: `RELEASE=update-me;tar -zcvf ${RELEASE}.tar.gz -s /config/${RELEASE}/ config/`
-1. Once everything has settled, tag and push the release with `git tag $VERSION` and `git push upstream $VERSION`. You can also tag the release in the GitHub UI.
-   - The `modelmesh-controller` image will be published via GitHub Actions.
-1. Upload generated install manifests to GitHub release assets.
-1. Be sure to create and push a new tag for the `kserve/modelmesh-minio-examples` image.
+It's generally a good idea to search the repo or `control`-`shift`-`f` for strings
+of the old version number and replace them with the new, keeping in mind conflicts
+with other library version numbers.
+
+Some of the steps below need to be performed at least twice:
+
+- once for the release candidate(s) (`v0.10.0-rc0`, `v0.10.0-rc1`, ...) and
+- once more for the actual release (`v0.10.0`).
+
+1. Create new (pre-)release tags (v...`-rc0`) in these repositories:
+
+   - [ ] [`modelmesh`](https://github.com/kserve/modelmesh/releases)
+   - [ ] [`modelmesh-runtime-adapter`](https://github.com/kserve/modelmesh-runtime-adapter/releases)
+   - [ ] [`rest-proxy`](https://github.com/kserve/rest-proxy/releases)
+
+   using the newly created `release-*` branches as target. This can be done by
+   creating a draft release using the GitHub web interface and checking the
+   "Set as a pre-release" option. The newly created tag should trigger the GitHub
+   action to push the respective (pre-)release container images to DockerHub which
+   are needed in the next step.
+
+2. Build and push a new tag for the `kserve/modelmesh-minio-examples` image. If
+   there have been no changes since the last release, you can create a new tag
+   from the previous release, e.g. creating a new `v0.10.0` tag from `latest`:
```

Comment:
I guess we'll link to the minio build doc here once that's merged?

Reply:
yes --we could leave this PR in draft for now until we have things settled -- or do a later update -- this instruction still works as is -- although not ideal